clients/js-legacy/src/actions.ts

```diff
@@ -0,0 +1,272 @@
+ * @param record          Record account address
+ * @param authority       Record account authority
+ * @param recordSize      Size of the record to be stored
+ * @param confirmOptions  Options for confirming the transaction
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function initializeRecord(
+    connection: Connection,
+    payer: Signer,
+    record: Signer,
+    authority: PublicKey,
+    recordSize: number,
+    confirmOptions?: ConfirmOptions,
+) {
+    const programId = RECORD_PROGRAM_ID;
+    const recordAccountSize = RECORD_META_DATA_SIZE + recordSize;
+    const lamports = await connection.getMinimumBalanceForRentExemption(Number(recordAccountSize));
+    const transaction = new Transaction().add(
+        SystemProgram.createAccount({
+            fromPubkey: payer.publicKey,
+            newAccountPubkey: record.publicKey,
+            space: Number(recordAccountSize),
+            lamports,
+            programId,
+        }),
+        createInitializeInstruction(
+            record.publicKey,
+            authority,
+        )
+    );
+    return await sendAndConfirmTransaction(connection, transaction, [payer, record], confirmOptions);
+}
+
+/**
+ * Write to an initialized record account
+ *
+ * @param connection      Connection to use
+ * @param payer           Payer of the transaction fees
+ * @param record          Record account address
+ * @param authority       Record account authority keypair
+ * @param recordSize      Offset to store the record bytes
+ * @param buffer          Record bytes
+ * @param confirmOptions  Options for confirming the transaction
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function writeRecord(
+    connection: Connection,
+    payer: Signer,
+    record: PublicKey,
+    authority: Signer,
+    offset: number,
+    buffer: Uint8Array,
+    confirmOptions?: ConfirmOptions,
+) {
+    let transactionResults = [];
+    let bufferOffset = 0;
+    let remainingBuffer = buffer;
+    while (bufferOffset < remainingBuffer.length) {
+        const currentChunkBuffer = remainingBuffer.subarray(
+            bufferOffset,
+            bufferOffset + RECORD_CHUNK_SIZE_POST_INITIALIZE
+        );
+        const transaction = new Transaction().add(
+            createWriteInstruction(
+                record,
+                authority.publicKey,
+                offset + bufferOffset,
+                currentChunkBuffer,
+            )
+        );
+        transactionResults.push(sendAndConfirmTransaction(connection, transaction, [payer, authority], confirmOptions));
+        bufferOffset = bufferOffset + RECORD_CHUNK_SIZE_POST_INITIALIZE;
+    }
+    return await Promise.all(transactionResults);
+}
+
+/**
+ * Initialize and write to a record account
+ *
+ * @param connection      Connection to use
+ * @param payer           Payer of the transaction fees
+ * @param record          Record account address
```

Comment:
nit: keypair here too
```suggestion
 * @param record          Record account signer
```